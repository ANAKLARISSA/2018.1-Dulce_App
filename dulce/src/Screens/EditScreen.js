import React from 'react';
import {View, TextInput, Text} from 'react-native';
import AGRButton from '../Components/AGRButton';


const styles = {
  container:{
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF',
    paddingHorizontal: 20,
  },
  input:{
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    alignSelf: 'center',
    width: '80%',
    marginTop: 10,
  },
  button: {
    alignItems: 'center',
    backgroundColor: 'pink',
    padding: 10,
    color: 'blue'
  },
};

class EditScreen extends React.Component {
  constructor(props){
    super(props);
    this.state = {editable:false};
  }

  tornarVisivel(){
    this.setState({
      editable: !this.state.editable
    });
  }


  render() {
    return (
      <View style={styles.container}>
        <TextInput style={styles.input}
          placeholder='Nome'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Matricula'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Hospital'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Setor'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Editar senha'
          editable = {this.state.editable}
        />
        <AGRButton
          text ="Editar perfil"
          style={styles.button}
          onPress={() => {this.tornarVisivel()}}
        />

      </View>
    );
  }
}


export default EditScreen;
