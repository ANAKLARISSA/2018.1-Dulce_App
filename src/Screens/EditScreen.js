import React from 'react';
import {View} from 'react-native';
import ScreenHeader from '../Components/ScreenHeader';
import ValidationComponent from 'react-native-form-validator';
//import store from '../Reducers/store';
//import axios from 'axios';
import {Container} from 'native-base';
import t from 'tcomb-form-native';
import SignUpButton from '../Components/SignUpButton';

const Form = t.form.Form;

const styles = {
  container: {
    justifyContent: 'center',
    marginTop: 10,
    padding: 15,
    backgroundColor: '#FFF'
  },
  text: {
    fontSize: 15,
    marginBottom: 10,
    marginLeft: 10,
    color: 'black',
    fontFamily: 'Raleway'
  },
  button: {
    height: 45,
    paddingHorizontal: 50,
    backgroundColor: '#5f4b8b',
    alignSelf: 'center',
    alignItems: 'center',
    width: 145
  }
};


export default class EditScreen extends ValidationComponent {
  constructor(props) {
    super(props);
    this.Service = t.struct({
      nome: t.String,
      email: t.String,
      senha: t.String,
      confirmarSenha: t.String
    });
    this.state = {
      value: default_state
    };
    this.options = {
      fields: default_field_options,
      stylesheet: formStyles
    };
  }

  onChange(value) {
    this.setState({value});
  }
  render() {
    return (
<<<<<<< HEAD
      <Container>
        <ScreenHeader title='Editar conta' />
        <View style={styles.container}>
          <Form
            ref='form'
            options={this.options}
            value={this.state.value}
            type={this.Service}
            onChange={(v) => this.onChange(v)}
          />
        </View>
        <SignUpButton
          text = 'Editar'
          onPress = {() => {}}
          style = {styles.button}
        />
      </Container>
=======
      <View style={styles.container}>
      <ScreenHeader title='Editar Perfil' icon='arrow-back'/>
        <ScrollView>
          <SmallLogo/>
          <AGRInput style={styles.input} value = {this.state.name} editable = {false} nameLabel='Nome'/>
          {this.isFieldInError('name') && this.getErrorsInField('name').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} value = {this.state.registration} editable = {false}
            placeholder='Matricula' nameLabel='Matrícula'/>
          <AGRInput value = {this.state.hospital} style={styles.input} editable = {this.state.editable}
            nameLabel='Hospital' onChangeText={(text) => this.setState({hospital: text})}/>
          {this.isFieldInError('hospital') && this.getErrorsInField('hospital').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} value = {this.state.sector} editable = {this.state.editable}
            nameLabel='setor' onChangeText={(text) => this.setState({sector: text})}/>
          {this.isFieldInError('sector') && this.getErrorsInField('sector').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} editable = {this.state.editable} nameLabel='Editar senha'
            secureTextEntry onChangeText={(text) => this.setState({password: text})}/>
          {this.isFieldInError('password') && this.getErrorsInField('password').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <View style={styles.alinhar}>
            <AGRButton style={styles.button} text='Salvar' onPress={() => {this.save(); this._onPressButton();} }/>
          </View>
        </ScrollView>
      </View>
>>>>>>> 9930bc4... #386 Adicionando botões nas telas
    );
  }
}
const formStyles = {
  ...Form.stylesheet,
  error: {
    color: 'red',
    fontSize: 14,
    marginBottom: 3,
    fontWeight: '600'
  }
};

const default_field_options = {
  Nome: {
    error: 'Campo obrigatório'
  },
  confirmarSenha: {
    secureTextEntry: true,
    error: 'Campo obrigatório'
  },
  Email: {
    error: 'Campo obrigatório'
  },
  senha: {
    secureTextEntry: true,
    error: 'Campo obrigatório'
  }
};

const default_state = {
  nome: '',
  email: '',
  senha: '',
  confirmarSenha: ''
};
