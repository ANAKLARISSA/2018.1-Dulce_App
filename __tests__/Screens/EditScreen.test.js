import 'react-native';
import React from 'react';
import EditScreen from '../../src/Screens/EditScreen';
import Adapter from 'enzyme-adapter-react-16';
//import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
//import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import t from 'tcomb-form-native';

require('bezier');

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
    .create(<EditScreen />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('test onChange value',() => {
  const wrapper = shallow(<EditScreen/>);
  const onChange = wrapper.instance().onChange();
  console.log(onChange);
});

it('testing onChange container',() => {
  const type = t.struct({
    nome: t.String,
    email: t.String,
    senha: t.String,
    confirmarSenha: t.String
  });
  const wrapper = shallow(<EditScreen type={type}/>);
  const container = wrapper.find('Styled(Container)').at(0);
  const form = container.find('Form').at(0);
  form.props().onChange();
});

//const flushPromises = () => new Promise(resolve => setImmediate(resolve));
//
// it('testing axios', async () => {
//   const wrapper = shallow(<EditScreen />);
//   await flushPromises();
//   wrapper.update();
//   expect(wrapper).toMatchSnapshot();
// });
