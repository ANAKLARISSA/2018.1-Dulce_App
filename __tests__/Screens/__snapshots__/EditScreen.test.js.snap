// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fd7d01",
        "flex": 0.25,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f1f1f1",
          "borderRadius": 5,
          "height": 40,
          "margin": 1,
          "width": 7.5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 8,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderBottomColor": "#a7a6ab",
              "borderBottomWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 15,
              "right": 0,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "top": 0,
            },
            Object {
              "backgroundColor": "#5f4b8b",
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 6,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 30,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 0,
                      "paddingTop": 2,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              virtual={undefined}
            >
              Ôèè
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {},
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "Raleway",
                },
              ]
            }
            virtual={undefined}
          >
             
            Editar conta
             
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "height": 1334,
          },
          Object {
            "backgroundColor": "#FFF",
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "justifyContent": "center",
            "marginTop": 10,
            "padding": 15,
          }
        }
      >
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#1a1515",
                  "fontFamily": "Lato",
                  "fontSize": 17,
                  "fontWeight": "200",
                  "marginBottom": 7,
                }
              }
            >
              Nome
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Nome"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={undefined}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 20,
                    "borderWidth": 1,
                    "color": "#1a1515",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#1a1515",
                  "fontFamily": "Lato",
                  "fontSize": 17,
                  "fontWeight": "200",
                  "marginBottom": 7,
                }
              }
            >
              Sobrenome
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Sobrenome"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={undefined}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 20,
                    "borderWidth": 1,
                    "color": "#1a1515",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#1a1515",
                  "fontFamily": "Lato",
                  "fontSize": 17,
                  "fontWeight": "200",
                  "marginBottom": 7,
                }
              }
            >
              Email
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Email"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={undefined}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 20,
                    "borderWidth": 1,
                    "color": "#1a1515",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            undefined,
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#5f4b8b",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 57,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 2,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "marginTop": "80%",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingHorizontal": 30,
                "paddingTop": 6,
                "width": 180,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 25,
                    },
                    Object {
                      "paddingHorizontal": 15,
                    },
                    Object {
                      "fontFamily": "Entypo",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                Ôàπ
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignItems": "center",
                    "color": "white",
                    "fontFamily": "Raleway",
                    "fontSize": 16,
                    "marginRight": 40,
                  }
                }
              >
                Salvar
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
